### PR TITLE
feat: migrate legacy auth request to approver ux

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out Leather build f48c5af — [Extension build](https://github.com/leather-io/extension/actions/runs/12202376262), [Test report](https://leather-io.github.io/playwright-reports/feat/use-approver-legacy-auth-request), [Storybook](https://feat/use-approver-legacy-auth-request--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=feat/use-approver-legacy-auth-request)

Investigating how we can keep other wallet options & use our new approver UX, realised the easiest option is to update it in the wallet. This wasn't too much work, versus the changes required lockstack's side.

cc/ @camerow, we'll want to undo the change discussed yesterday.

https://github.com/user-attachments/assets/1effaad6-5325-4666-8940-6d7f94b71005

Don't pay attention to slow loading icon 👀

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -25,16 +25,13 @@ module.exports = {
     'plugin:storybook/csf',
   ],
   ignorePatterns: ['./leather-styles'],
-  plugins: ['react', 'react-hooks', '@typescript-eslint', 'deprecation'],
```
Comment (kyranjamie) [inline .eslintrc.cjs:28]: The deprecation plugin is itself deprecated as its included in ts-eslint rules. Also throwing incorrect chrome deprecation error.

**Review by alter-eggo — APPROVED** [review]

**Comment by camerow** [comment]
Thanks Kyran, I'll undo the updates we made to the authprovider on lockstacks

**Review by camerow — APPROVED** [review]

**Comment by markmhendrickson** [comment]
Any more code review or QA needed or should we get this merged and released?

**Comment by kyranjamie** [comment]
Let's do a QA on the release PR. cc/ @314159265359879 if you'll be able to help with this. This PR changes the auth flow to new UI, so testing this is main concern. Our tests pass now, which should cover most bases, though (as many subsequent tests rely on legacy auth).

**Comment by markmhendrickson** [comment]
I just tested with Lockstacks and Gamma (Stacks) and it worked great 💪

**Comment by 314159265359879** [comment]
@kyranjamie With Leather in hardware mode there is an issue. I am having trouble getting the lockstacks dapp to complete the connection. I see this error in the console (lockstacks):
image
wallet remains unconnected to lockstacks
image

On gamma connecting also fails: 

In both cases this is only thing displayed in the leather pop-over console:
image

I notice that no error is displayed about unlocking the ledger device when it is still locked. I assume that tells us something about when the connection with the device fails/isn't created?



With Leather in software mode it is working perfectly. I also tested and looked at the behavior of the new auth flow UI when I switch accounts on the extension, and it behaves exactly as expected when I use a software wallet.

**Comment by 314159265359879** [comment]
https://github.com/user-attachments/assets/16a45e95-fc07-4671-9634-6469bc0d2da0

**Comment by kyranjamie** [comment]
I'll look into this now, thanks @314159265359879

**Pull request merged**